netbox/utilities/templates/buttons/add.html

```diff
@@ -1,6 +1,10 @@
 {% if url %}
-{% load i18n %}
-  <a href="{{ url }}" type="button" class="btn btn-primary">
-    <i class="mdi mdi-plus-thick"></i> {% trans "Add" %}
-  </a>
-{% endif %}
+    {% load i18n %}
+    {% if return_url %}
+        <a href="{{ url }}?return_url={{ return_url }}" type="button" class="btn btn-primary">
+    {% else %}
+        <a href="{{ url }}" type="button" class="btn btn-primary">
+    {% endif %}
+<i class="mdi mdi-plus-thick"></i> {% trans "Add" %}
+</a>
+{% endif %}
```

Comment:
This can be simplified to avoid repeating ourselves:
```suggestion
{% if url %}
{% load i18n %}
  <a href="{{ url }}{% if return_url %}?return_url={{ return_url }}{% endif %}" type="button" class="btn btn-primary">
    <i class="mdi mdi-plus-thick"></i> {% trans "Add" %}
  </a>
{% endif %}
```